pkg/adaptor/cloud/aws/provider.go

```diff
@@ -168,6 +164,10 @@ func (p *awsProvider) CreateInstance(ctx context.Context, podName, sandboxID str
 		b64EncData = base64.StdEncoding.EncodeToString([]byte(userData))
 	}
 
+	if err = p.updateInstanceTypeSpecList(); err != nil {
+		return nil, err
+	}
+
```

Comment:
This method is time consuming and calling it for every CreateInstance call will be suboptimal. 
Populating the instance spec should be during initialisation imho.

Reply:
do you have an alternative idea?

Reply:
Not yet :-(